Components/Form.js

```diff
@@ -0,0 +1,73 @@
+import React, { useState } from "react";
+
+function Form() {
+  const [todo, setTodo] = useState("");
+  const [todoList, setTodoList] = useState([]);
+  
+  const handleChange = (e) => {
+    setTodo(e.target.value);
+  };
+
+  const addTodo = () => {
+    if (todo != "") {
+      const todoInsert = {
+        id: Math.floor(Math.random() * 10000), //Generate random id from 0 to 10000
+        value: todo,
+        isCompleted: false,
+      };
+
+      setTodoList([...todoList, todoInsert]); //... operator "spreads out the own enumerable properties as properties of the element you're creating" stack overflow
+      setTodo(" ");
+    }
+  };
+
+  const deleteTodo = (e, id) => {
+    e.preventDefault();
+    setTodoList(todoList.filter((t) => t.id != id));
+  };
+
+  const completeTodo = id => {
+   setTodoList(todoList.map(item => {
+       if(item.id == id){
+        item.isCompleted = !item.isCompleted   
+       }
+       return item  
+     })
+   )
+  }
+
+  return (
+    <div>
+      <label htmlFor="newItem">Nuovo item</label>
+      <input
+        type="text"
+        name="newItem"
+        id="newItem"
+        onChange={(e) => handleChange(e)}
+        placeholder="Insert todo"
+      />
+      <button onClick={addTodo}>Add</button>
+
+      {todoList.length > 0 ? (
+        <ul>
+          {todoList.map((t) => (
+            <div className="todoLi" key={t.id}>
+              <li>
+                {t.value}
+                <button onClick={(e) => deleteTodo(e, t.id)}>Delete</button>
+                <input
+                  type="checkbox"
+                  onChange={() => completeTodo(t.id)}
+                  checked={t.isCompleted}
+                />
+              </li>
+              {console.log(todoList)}
```

Comment:
A better way to log data for debug purposes is to use a `useEffect` hook. Here's an example of it:

```js
React.useEffect(() => {
  console.log("Todos:", todoList);
}, [todoList]);
```

The callback runs each time the `todoList` object changes (make sure that the dependencies array contains the prop and states that you are using inside the callback).